### PR TITLE
Adding default templates for Metricbeat ECS data

### PR DESCRIPTION
**Pull request by jbaiera**
This PR adds four new templates that are automatically installed from the Monitoring plugin.

In 8.x, Metricbeat will be writing its data in ECS compliant format, even when used with xpack mode enabled (stack monitoring). In order to continue to support the legacy data format, new mappings have been created with the new ECS fields for indexing data, and alias fields for the legacy format which point to the corresponding ECS fields.

New templates added:
* Template `.monitoring-beats-mb` matching on `.monitoring-beats-8-*` defined in `monitoring-beats-mb.json`
* Template `.monitoring-es-mb` matching on `.monitoring-es-8-*` defined in `monitoring-es-mb.json`
* Template `.monitoring-kibana-mb` matching on `.monitoring-kibana-8-*` defined in `monitoring-kibana-mb.json`
* Template `.monitoring-logstash-mb` matching on `.monitoring-logstash-8-*` defined in `monitoring-logstash-mb.json`

All templates are being created with internal version `8000099`

### CODE REVIEW

**Comment by jbaiera** [comment]
@elasticmachine update branch

**Review by klacabane — APPROVED** [review]

**Comment by jbaiera** [comment]
@elasticmachine run elasticsearch-ci/part-2

File: docs/reference/monitoring/indices.asciidoc [hunk 1]
```diff
@@ -28,7 +28,7 @@ and the number of replicas to two.
 ----------------------------------
 PUT /_template/custom_monitoring
 {
-  "index_patterns": ".monitoring-*",
+  "index_patterns": [".monitoring-beats-7*", ".monitoring-es-7*", ".monitoring-kibana-7*", ".monitoring-logstash-7*"],
```
Comment (matschaffer) [inline docs/reference/monitoring/indices.asciidoc:31]: Pretty sure this is more in line with what's in the template file.

```suggestion
  "index_patterns": [".monitoring-beats-7-*", ".monitoring-es-7-*", ".monitoring-kibana-7-*", ".monitoring-logstash-7-*"],
```

File: x-pack/plugin/core/src/main/resources/monitoring-beats-mb.json [hunk 2]
```diff
@@ -0,0 +1,2198 @@
+{
+  "index_patterns": [".monitoring-beats-${xpack.stack.monitoring.template.version}-*"],
```
Comment (matschaffer) [inline x-pack/plugin/core/src/main/resources/monitoring-beats-mb.json:2]: This is the same pattern as https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/core/src/main/resources/monitoring-es.json#L2 - are we planning to introduce a new variable to set the mb version to `10`?

**Review by matschaffer — COMMENTED** [review]
Some read-through observations, still don't quite have my head wrapped around all this.

File: x-pack/plugin/core/src/main/resources/monitoring-beats-mb.json (hunk 2 above)
Comment (jbaiera) [reply x-pack/plugin/core/src/main/resources/monitoring-beats-mb.json:2]: Currently the `xpack.stack.monitoring.template.version` is set to version 8 for the metricbeat indices. We talked it over a bit more and agreed that 8 was indeed the more appropriate version number to put here. The monitoring-es.json file uses the `xpack.monitoring.template.version` property (doesn't contain `stack`)

**Comment by jbaiera** [comment]
@elasticmachine update branch

File: x-pack/plugin/core/src/main/resources/monitoring-beats-mb.json (hunk 2 above)
Comment (matschaffer) [reply x-pack/plugin/core/src/main/resources/monitoring-beats-mb.json:2]: ah... `xpack.monitoring.template.version` vs `xpack.stack.monitoring.template.version` - I missed the distinction. Thanks!

**Pull request merged**